### ISSUE TITLE
copy.deepcopy on JAX params is expensive — use jax.tree.map

### ISSUE THREAD

**Issue by zeinali72** [zeinali72/swe-pinn#59]
## Description

In `src/training/loop.py`, lines 165 and 175:

```python
best_params_nse = copy.deepcopy(params)
best_params_loss = copy.deepcopy(params)
```

`copy.deepcopy` on JAX arrays copies device memory to host, duplicates, and copies back.

## Impact

For large models (512-wide, 6-deep), this adds 10-50ms per checkpoint update, twice per epoch when both best-NSE and best-loss improve.

## Fix

```python
best_params_nse = jax.tree.map(jnp.copy, params)
```

## Category
**Medium** · Pillar: Performance Check
